python/tests/avro/test_reader.py

```diff
@@ -0,0 +1,347 @@
+                        7: 19513,
+                        8: 19513,
+                        9: 19513,
+                        10: 19513,
+                        11: 19513,
+                        12: 19513,
+                        13: 19513,
+                        14: 19513,
+                        15: 19513,
+                        16: 19513,
+                        17: 19513,
+                        18: 19513,
+                        19: 19513,
+                    },
+                    {
+                        1: 19513,
+                        2: 0,
+                        3: 0,
+                        4: 19513,
+                        5: 19513,
+                        6: 19513,
+                        7: 0,
+                        8: 0,
+                        9: 19513,
+                        10: 0,
+                        11: 0,
+                        12: 19513,
+                        13: 0,
+                        14: 0,
+                        15: 0,
+                        16: 0,
+                        17: 0,
+                        18: 0,
+                        19: 0,
+                    },
+                    {16: 0, 17: 0, 18: 0, 19: 0, 10: 0, 11: 0, 12: 0, 13: 0, 14: 0, 15: 0},
+                    {
+                        2: b"2020-04-01 00:00",
+                        3: b"2020-04-01 00:12",
+                        7: b"\x03\x00\x00\x00",
+                        8: b"\x01\x00\x00\x00",
+                        10: b"\xf6(\\\x8f\xc2\x05S\xc0",
+                        11: b"\x00\x00\x00\x00\x00\x00\x00\x00",
+                        13: b"\x00\x00\x00\x00\x00\x00\x00\x00",
+                        14: b"\x00\x00\x00\x00\x00\x00\xe0\xbf",
+                        15: b")\\\x8f\xc2\xf5(\x08\xc0",
+                        16: b"\x00\x00\x00\x00\x00\x00\x00\x00",
+                        17: b"\x00\x00\x00\x00\x00\x00\x00\x00",
+                        18: b"\xf6(\\\x8f\xc2\xc5S\xc0",
+                        19: b"\x00\x00\x00\x00\x00\x00\x04\xc0",
+                    },
+                    {
+                        2: b"2020-04-30 23:5:",
+                        3: b"2020-05-01 00:41",
+                        7: b"\t\x01\x00\x00",
+                        8: b"\t\x01\x00\x00",
+                        10: b"\xcd\xcc\xcc\xcc\xcc,_@",
+                        11: b"\x1f\x85\xebQ\\\xe2\xfe@",
+                        13: b"\x00\x00\x00\x00\x00\x00\x12@",
+                        14: b"\x00\x00\x00\x00\x00\x00\xe0?",
+                        15: b"q=\n\xd7\xa3\xf01@",
+                        16: b"\x00\x00\x00\x00\x00`B@",
+                        17: b"333333\xd3?",
+                        18: b"\x00\x00\x00\x00\x00\x18b@",
+                        19: b"\x00\x00\x00\x00\x00\x00\x04@",
+                    },
+                    None,
+                    [4],
+                    0,
+                ]
+            ),
+        ]
+    )
+
+
+def test_singleton():
```

Comment:
Should this be moved to a different suite? Seems strange to include it here in the reader tests.

Reply:
I've moved it to `test_singleton` 👍🏻